### ISSUE TITLE
GWC tile rendering fails when layer uses workspace style

### ISSUE THREAD

**Issue by srstsavage** [GeoWebCache/geowebcache#232]
(Originally discovered by Mark Volz, see [mailing list message](http://sourceforge.net/p/geoserver/mailman/message/31957143/)).

GWC fails to generate tiles for GeoServer workspace layers which use a workspace bound style. GWC presumably uses the global WMS endpoint to generate tiles. Since workspace styles aren't visibly in the global endpoints, rendering fails due to not finding the workspace style.

GWC throws the following:

```
08 Apr 11:18:11 ERROR [seed.MTSeeder] - Problem communicating with GeoServer
org.geowebcache.GeoWebCacheException: Problem communicating with GeoServer
       at org.geoserver.gwc.layer.GeoServerTileLayer.getMetatilingReponse(GeoServerTileLayer.java:522)
       at org.geoserver.gwc.layer.GeoServerTileLayer.seedTile(GeoServerTileLayer.java:711)
       at org.geowebcache.seed.SeedTask.doActionInternal(SeedTask.java:136)
       at org.geowebcache.seed.GWCTask.doAction(GWCTask.java:76)
       at org.geowebcache.seed.MTSeeder.call(MTSeeder.java:36)
       at org.geowebcache.seed.MTSeeder.call(MTSeeder.java:25)
       at java.util.concurrent.FutureTask.run(Unknown Source)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
       at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalStateException: Expected: RenderedImageMap, got null
       at org.geoserver.gwc.layer.GeoServerTileLayer.dispatchGetMap(GeoServerTileLayer.java:575)
       at org.geoserver.gwc.layer.GeoServerTileLayer.getMetatilingReponse(GeoServerTileLayer.java:517)
       ... 9 more
```

and previously in the logs, the underlying GetMap exception:

```
08 Apr 11:18:11 ERROR [geoserver.ows] - 
org.geoserver.platform.ServiceException: No such style: the_style
        at org.geoserver.wms.map.GetMapKvpRequestReader.parseStyles(GetMapKvpRequestReader.java:1386)
        at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:318)
        at org.geoserver.wms.map.GetMapKvpRequestReader.read(GetMapKvpRequestReader.java:80)
        at org.geoserver.ows.Dispatcher.parseRequestKVP(Dispatcher.java:1412)
        at org.geoserver.ows.Dispatcher.dispatch(Dispatcher.java:622)
        at org.geoserver.ows.Dispatcher.handleRequestInternal(Dispatcher.java:263)
        at org.springframework.web.servlet.mvc.AbstractController.handleRequest(AbstractController.java:153)
        at org.geoserver.gwc.GWC.dispatchOwsRequest(GWC.java:1162)
        at org.geoserver.gwc.layer.GeoServerTileLayer.dispatchGetMap(GeoServerTileLayer.java:572)
        at org.geoserver.gwc.layer.GeoServerTileLayer.getMetatilingReponse(GeoServerTileLayer.java:517)
        at org.geoserver.gwc.layer.GeoServerTileLayer.seedTile(GeoServerTileLayer.java:711)
        at org.geowebcache.seed.SeedTask.doActionInternal(SeedTask.java:136)
        at org.geowebcache.seed.GWCTask.doAction(GWCTask.java:76)
        at org.geowebcache.seed.MTSeeder.call(MTSeeder.java:36)
        at org.geowebcache.seed.MTSeeder.call(MTSeeder.java:25)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
```

GeoServer specs:

Version
2.5-SNAPSHOT
Git Revision
af402286a9b28dd661581db66880b77e53a2f64a
Build Date
20-Mar-2014 15:30
GeoTools Version
11-SNAPSHOT (rev f1d3a03745347ec7246b4cdf61b385dc97d3c5a1)
GeoWebCache Version
1.5-SNAPSHOT (rev 4e59f43099865dc68cb211e6df8027c5f73b8beb/4e59f43099865dc68cb211e6df8027c5f73b8beb)

**Comment by mapyourproperty** [comment]
So the temporary solution is move all your style out the workspace and into the global workspace area.  I.e. when create style do not specify a workspace.
In detail, I noticed that if I used a default style like "polygon", GWC would work in seeding.  This is because the default style "polygon' is in the global workspace.

**Comment by smithkm** [comment]
The best way to make this work at the moment is to add a `WORKSPACE` parameter filter with the name of the workspace as the default value.  

The relevant GeoServer bug tickets are  [GEOS-6172](https://jira.codehaus.org/browse/GEOS-6172) and [GEOS-6512](https://jira.codehaus.org/browse/GEOS-6512)

GWC feature requests that would make this easier to fix in GeoServer are #170 and #257

**Comment by jometho** [comment]
Hi @smithkm, I'm having a similar issue, I have moved the layer style to the same directory as the layer from global workspace and still nothing works.

What was the solution found for this problem? How did you solve yours @shprabin ?
